jib-core/src/main/java/com/google/cloud/tools/jib/registry/ManifestPusher.java

```diff
@@ -85,8 +92,28 @@ public Void handleHttpResponseException(HttpResponseException httpResponseExcept
   }
 
   @Override
-  public Void handleResponse(Response response) {
-    return null;
+  public DescriptorDigest handleResponse(Response response)
+      throws IOException, UnexpectedImageDigestException {
+    // Checks if the image digest is as expected.
+    DescriptorDigest expectedDigest =
+        JsonTemplateMapper.toBlob(manifestTemplate)
+            .writeTo(ByteStreams.nullOutputStream())
+            .getDigest();
+
+    List<String> receivedDigests = response.getHeader(RESPONSE_DIGEST_HEADER);
+    if (receivedDigests.size() == 1) {
```

Comment:
Any chance the server returns multiple headers for some reason and one of the values is the expected match?

Reply:
I would not think so, at least the spec indicates that only one header would be returned